Bottom: b6abbae3ed03a0cab897e36b7db93d517b3c6e7e
Top:    3b74d770ea3023781fd6c6317695fd13aa927259
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-06-27 14:20:06 +0100

bdale patch


---

diff --git a/src/action.c b/src/action.c
index 8b7e6d5..164a8f9 100644
--- a/src/action.c
+++ b/src/action.c
@@ -7999,6 +7999,71 @@ ActionAttributes (int argc, char **argv, int x, int y)
   return 0;
 }
 
+
+static const char outline_to_poly_syntax[] =
+"OutlineToPoly()";
+
+static const char outline_to_poly_help[] =
+"Convert PCB's own interal idea of the outline shape (as a polygon), to a"
+"polygon on the \"outline\" layer, deleting all other content there\n";
+
+/* %start-doc actions OutlineToPoly
+
+Test for BDale
+
+%end-doc */
+
+
+static int
+ActionOutlineToPoly (int argc, char **argv, int x, int y)
+{
+  char *function = ARG (0);
+  POLYAREA *board_outline;
+  LayerType *Layer = NULL;
+  int found_outline;
+  int i;
+
+  printf ("Hello world!\n");
+
+  for (i = 0; i < max_copper_layer; i++)
+    {
+      Layer = PCB->Data->Layer + i;
+
+      if (strcmp (Layer->Name, "outline") == 0 ||
+          strcmp (Layer->Name, "route") == 0)
+        {
+          found_outline = 1;
+          break;
+        }
+    }
+
+  if (!found_outline) {
+    printf ("Didn't find outline\n");
+    return 0;
+  }
+
+
+  board_outline = board_outline_poly ();
+
+  PolyToPolygonsOnLayer (PCB->Data, Layer, board_outline, MakeFlags (0));
+
+  LINE_LOOP (Layer);
+  {
+    RemoveLine (Layer, line);
+  }
+  END_LOOP;
+
+  ARC_LOOP (Layer);
+  {
+    RemoveArc (Layer, arc);
+  }
+  END_LOOP;
+
+  Draw ();
+
+  return 1;
+}
+
 /* --------------------------------------------------------------------------- */
 
 HID_Action action_action_list[] = {
@@ -8190,6 +8255,9 @@ HID_Action action_action_list[] = {
   {"Import", 0, ActionImport,
    import_help, import_syntax}
   ,
+  {"OutlineToPoly", 0, ActionOutlineToPoly,
+   outline_to_poly_help, outline_to_poly_syntax}
+  ,
 };
 
 REGISTER_ACTIONS (action_action_list)
diff --git a/src/polygon.c b/src/polygon.c
index 5875a0f..d031725 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1967,8 +1967,8 @@ struct clip_outline_info {
   POLYAREA *poly;
 };
 
-#define ROUTER_THICKNESS 1000
-//#define ROUTER_THICKNESS 10
+//#define ROUTER_THICKNESS 1000
+#define ROUTER_THICKNESS 10
 
 static int
 arc_outline_callback (const BoxType * b, void *cl)
